src/foam/nanos/auth/Address.js

```diff
@@ -219,27 +223,326 @@ foam.CLASS({
               e.NEQ(foam.nanos.auth.Address.COUNTRY_ID, 'US'),
               e.REG_EXP(
                 foam.nanos.auth.Address.POSTAL_CODE,
-                /^^\d{5}(?:[-\s]\d{4})?$/i)
+                /^\d{5}(?:[-\s]\d{4})?$/i)
+            );
+          },
+          errorMessage: 'INVALID_POSTAL_CODE_ERR_MSG'
+        },
+        // Austria
+        {
+          args: ['postalCode', 'countryId'],
+          predicateFactory: function(e) {
+            return e.OR(
+              e.NEQ(foam.nanos.auth.Address.COUNTRY_ID, 'AT'),
```

Comment:
@kgrgreer can we use mmethod here for switching on country code to get the appropriate reg-ex?

Reply:
No, mmethod only works with types. We could make a switch/case mlang to make this easier, but we don't current have such a thing.